src/common/ServerPushSetupBanner.js

```diff
@@ -0,0 +1,88 @@
+/* @flow strict-local */
+
+import React from 'react';
+import type { Node } from 'react';
+import isAfter from 'date-fns/isAfter';
+import subWeeks from 'date-fns/subWeeks';
+
+import ZulipBanner from './ZulipBanner';
+import { useSelector, useGlobalSelector, useDispatch } from '../react-redux';
+import { getIdentity, getAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getRealm, getGlobalSettings } from '../directSelectors';
+import { dismissServerPushSetupNotice } from '../account/accountActions';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying the server hasn't enabled push notifications, if so
+ *
+ * When dismissed, reappears again in two weeks.
+ */
+export default function PushNotifsSetupBanner(props: Props): Node {
+  const dispatch = useDispatch();
+  const lastDismissedServerPushSetupNotice = useSelector(
+    state => getAccount(state).lastDismissedServerPushSetupNotice,
+  );
+  const pushNotificationsEnabled = useSelector(state => getRealm(state).pushNotificationsEnabled);
+  const realm = useSelector(state => getIdentity(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+  const settings = useGlobalSelector(getGlobalSettings);
+
+  let visible = false;
+  let text = '';
+  if (pushNotificationsEnabled) {
+    // don't show
+  } else if (
+    lastDismissedServerPushSetupNotice !== null
+    && isAfter(
+      lastDismissedServerPushSetupNotice,
+      subWeeks(
+        // TODO: Could rerender this component on an interval, to give an
+        //   upper bound on how old this "current date" can be.
+        new Date(),
+        2,
+      ),
+    )
```

Comment:
nit: this whole comparison fits nicely on a line, if the comment is pulled to the outside of it:
```js
    // TODO: Could rerender this component on an interval, to give an
    //   upper bound on how old this "current date" can be.
    && isAfter(lastDismissedServerPushSetupNotice, subWeeks(new Date(), 2))
```

Needs a bit of rewording that way, because its physical location no longer points specifically to the `new Date()` it's about but rather to the whole comparison in general. But 7 lines of code turn into 1 line, and without a comment in the middle, making it much easier to take it all in at once. That's a good trade.